src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -368,12 +367,33 @@ protected virtual SamlConditions CreateConditions(SecurityTokenDescriptor tokenD
             else if (SetDefaultTimesOnTokenCreation)
                 conditions.NotOnOrAfter = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes);
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
-                conditions.Conditions.Add(new SamlAudienceRestrictionCondition(new Uri(tokenDescriptor.Audience)));
+            var uriList = CreateUriList(tokenDescriptor);
```

Comment:
You are creating a new list here when you don't really need to.

You could write
```
for( i=0; i<tokenDescriptor.Audiences;i++)
{
    if (!string.IsNullOrEmpty(tokenDescriptor.Audiences[i])
        samlAudienceRestrictionCondition.Audiences.Add(tokenDescriptor.Audiences[i])
}
```

Reply:
Fair point, ideally the public interface for SamlAudienceRestrictionCondition wouldn't make a new list if already presented with one but since that isn't the case it makes sense to instantiate and add audiences one at a time.